src/main/java/Duke.java

```diff
@@ -1,185 +1,46 @@
 
 public class Duke {
-    private static final String LINE = "____________________________________________________________\n";
     private static final String BYE = "bye";
     private static final String FILEPATH = "./data/tasks.txt";
-    private ArrayList<Task> inputList;
+    private TaskList tasks;
     private Storage storage;
+    private Ui ui;
 
     private Duke() {
-        inputList = new ArrayList<>();
+        ui = new Ui();
         storage = new Storage(FILEPATH);
-    }
-
-    public void init() {
-        Scanner sc = new Scanner(System.in);
         try {
-            inputList = storage.read();
+            tasks = new TaskList(storage.read());
         } catch (FileNotFoundException e) {
-            System.out.println(template("No existing user data file found. Welcome to the bot club:)"));
+            ui.showFileNotFound();
+            tasks = new TaskList();
         } catch (CorruptedFileException e) {
-            System.out.println("Your file has been corrupted :( Unfortunately the data cannot be used");
-        }
-        while (sc.hasNextLine()) {
-            String command = sc.nextLine();
-            if (command.equals(BYE)) {
-                System.out.println(template("Cya!!"));
-                return;
-            } else {
-                try {
-                    String reply = inputHandler(command);
-                    System.out.println(template(reply));
-                    storage.write(FILEPATH, inputList);
-                } catch (IllegalArgumentException | InvalidArgumentException | InvalidCommandException e){
-                    System.out.println(e.getMessage());
-                } catch (IOException e) {
-                    System.out.println("There is an issue writing to the storage file!!");
-                }
-            }
-        }
-    }
-
-    private String inputHandler(String input) throws InvalidArgumentException, InvalidCommandException {
-        String[] commands = input.split(" ", 2);
-        Commands current;
-        try {
-            current = Commands.valueOf(commands[0].toUpperCase());
-            switch(current) {
-            case LIST:
-                return display();
-            case DONE:
-                return updateTask(Integer.valueOf(commands[1]));
-            case TODO:
-                return addTodo(commands[1]);
-            case EVENT:
-                String[] arr = commands[1].split("/at");
-                return addEvent(arr[0], arr[1].trim());
-            case DEADLINE:
-                String[] arr2 = commands[1].split("/by");
-                return addDeadline(arr2[0], arr2[1].trim());
-            case DELETE:
-                return deleteTask(Integer.valueOf(commands[1]));
-            default:
-                throw new InvalidCommandException("");
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        ui.greet();
+        boolean isExit = false;
```

Comment:
Should "isExit" be "hasExit" instead? Perhaps it will sound more natural and grammatically correct if you intend for it to represent the state of Duke :)

Reply:
I use isExit as the program has not exited and hence, the variable represents "is exiting: instead of "has exited". Both sound pretty rough if were to look at it from a grammatical point of view :)